File: go-solutions/283_move_zeroes.go
package leetcode

func MoveZeroes(nums []int) []int {
	lastNonZero := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != 0 {
			nums[i], nums[lastNonZero] = nums[lastNonZero], nums[i]
			lastNonZero++
		}
	}
	return nums
}

File: ts-solutions/src/283-move-zeroes.ts
export function moveZeroes(nums: number[]): void {
	let lastNonZeroIndex = 0
	for (let i = 0; i < nums.length; i++) {
		if (nums[i] !== 0) {
			;[nums[i], nums[lastNonZeroIndex]] = [nums[lastNonZeroIndex], nums[i]]
			lastNonZeroIndex++
		}
	}
}
